org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/preferences/DefaultPersistentModel.java

```diff
@@ -53,6 +56,7 @@ public DefaultPersistentModel(IProject project, IPath buildDir, IPath buildScrip
         this.managedNatures = ImmutableList.copyOf(managedNatures);
         this.managedBuilders = ImmutableList.copyOf(managedBuilders);
         this.hasAutoBuildTasks = hasAutoBuildTasks;
+        this.gradleVersion = gradleVersion;
```

Comment:
I don't know this all very well, but hopefully this makes some sense...

If we're persisting the Gradle version used, what happens when this is out-of-sync with the actual Gradle build? e.g., I import with Gradle 5.0.  I manually change gradle-wrapper.properties to 5.6.  Are there cases where we'll interpret answers from a Gradle 5.6 version as if we were using Gradle 5.0 (or ask different questions because we think the version of Gradle is older) because the persisted value is different from the real value?

Reply:
That's correct, Buildship gets out of sync if you change the build (or import) configuration and don't call synchronize manually. This is a conscious decision: we don't want to keep track of files and eagerly call synchronize as it can be a very expensive operation that potentially blocks the user from continue working. The tasks view is a good example. If you add a new task, you have to explicitly click the refresh button to load the model to see the changes.

There's one problematic use-case though: updating from a previous version. We introduced a new entry in `PersistentModel` which will be incomplete if a user updates to the latest version (resulting in a runtime error). To overcome that, I've added a new method to the plugin startup to call synchronization for projects with an incomplete model (see `CorePlugin`).